### ISSUE TITLE
Race/cascade in DefaultInstanceProfileAWSCredentials

### ISSUE THREAD

**Issue by DTTerastar** [aws/aws-sdk-net#1625]
## Expected Behavior
Only one request to the metadata service should be made, even at startup

## Current Behavior
When a container starts up under load the code in https://github.com/aws/aws-sdk-net/blob/master/sdk/src/Core/Amazon.Runtime/Credentials/_bcl%2Bnetstandard/DefaultInstanceProfileAWSCredentials.cs#L88 allows every task/thread to call the metadata service.  This leads to rate limiting to kick in on the metadata service which then cascades down to other containers when they try and start or refresh the creds.

## Possible Solution
An double checked lock, Lazy or semphoreslim should be used to prevent every task/thread from making requests to IMDS

## Steps to Reproduce (for bugs)
This occurs in EKS under load testing conditions

## Context
This severely hurts load tests since adding capacity actually hurts the performance due to IMDS rate limiting

## Your Environment
* Service assembly and version used: 
```    
<PackageReference Include="AWSSDK.Core" Version="3.3.107.3" />
<PackageReference Include="AWSSDK.Extensions.NETCore.Setup" Version="3.3.101" />
<PackageReference Include="AWSSDK.S3" Version="3.3.111.4" />
```
* Operating System and version: 
FROM mcr.microsoft.com/dotnet/core/aspnet:3.1
* Visual Studio version: 
2019
* Targeted .NET platform: 
core 3.1

**Comment by DTTerastar** [comment]
Any news?  This is my workaround that fixes the issue for me: https://gist.github.com/DTTerastar/f373bb6dbfc8d1437665f7b23e0a2cbb

**Comment by DTTerastar** [comment]
I've opened sourced a small nuget library to help others fix this: https://github.com/AwsEx/AwsEx.Creds

**Comment by hunanniu** [comment]
This has been fixed in version 3.5.2 of AWSSDK.Core
